src/webots/nodes/WbWorldInfo.cpp

```diff
@@ -292,26 +297,31 @@ void WbWorldInfo::applyToOdeGlobalDamping() {
 
 // Computes an orthonormal basis whose 'yaw unit vector' is the opposite of the normalized gravity vector
 void WbWorldInfo::updateGravityBasis() {
-  if (!gravity().isNull()) {
-    mGravityUnitVector = gravity().normalized();
-    WbMathsUtilities::orthoBasis(-gravity(), mGravityBasis);
-  } else {
-    mGravityUnitVector.setXyz(0.0, -1.0, 0.0);
-    mGravityBasis[X].setXyz(1.0, 0.0, 0.0);
-    mGravityBasis[Y].setXyz(0.0, 1.0, 0.0);
-    mGravityBasis[Z].setXyz(0.0, 0.0, 1.0);
+  if (mCoordinateSystem->value() == "ENU") {
+    mEastVector = WbVector3(1, 0, 0);
+    mNorthVector = WbVector3(0, 1, 0);
+    mUpVector = WbVector3(0, 0, 1);
+    mGravityBasis[X].setXyz(0, 1, 0);
+    mGravityBasis[Y].setXyz(0, 0, 1);
+    mGravityBasis[Z].setXyz(1, 0, 0);
+  } else {  // "NUE"
+    mNorthVector = WbVector3(1, 0, 0);
+    mUpVector = WbVector3(0, 1, 0);
+    mEastVector = WbVector3(0, 0, 1);
+    mGravityBasis[X].setXyz(1, 0, 0);
+    mGravityBasis[Y].setXyz(0, 1, 0);
+    mGravityBasis[Z].setXyz(0, 0, 1);
   }
+  mGravityUnitVector = -mUpVector;
+  mGravityVector = mGravityUnitVector * mGravity->value();
 }
 
-void WbWorldInfo::updateNorthDirection() {
-  if (mNorthDirection->value().isNull()) {
-    mNorthDirection->setValue(1, 0, 0);
-    parsingWarn(tr("'northDirection' must be a unit vector. Reset to default value (1, 0, 0)."));
-  }
+void WbWorldInfo::updateCoordinateSystem() {
+  WbProtoTemplateEngine::setCoordinateSystem(mCoordinateSystem->value());
```

Comment:
This comment was never answered/addressed.

Reply:
That's correct, sorry for skipping it. I believe it will be difficult to regenerate all the procedural PROTOs and not very useful as changing the coordinate system shouldn't happen very often, so I will go for a warning.